File: app/src/main/java/com/alexkn/syntact/domain/common/TemplateType.java
package com.alexkn.syntact.domain.common;

import com.google.gson.annotations.SerializedName;

public enum TemplateType {

    @SerializedName("Basic")
    BASIC,

    @SerializedName("Advanced")
    ADVANCED,

    @SerializedName("Custom")
    CUSTOM
}

File: app/src/main/java/com/alexkn/syntact/data/common/TemplateType.kt
package com.alexkn.syntact.data.common

import com.google.gson.annotations.SerializedName

enum class TemplateType(val type: String) {

    @SerializedName(value = "Basic", alternate = ["basic", "BASIC"])
    BASIC("Basic"),

    @SerializedName(value = "Advanced", alternate = ["advanced", "ADVANCED"])
    ADVANCED("Advanced"),

    @SerializedName("Custom", alternate = ["custom", "CUSTOM"])
    CUSTOM("Custom")
}
